benchmark/distinct_user_id/nlohmann_json_sax.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#ifdef SIMDJSON_COMPETITION_NLOHMANN_JSON
+
+#include "distinct_user_id.h"
+#include <string.h>
+namespace distinct_user_id {
+
+using json = nlohmann::json;
+
+struct nlohmann_json_sax {
+    struct Handler : json::json_sax_t
+    {
+        std::vector<uint64_t>* result;
+        bool user = false;
+        bool user_id = false;
+        Handler(std::vector<uint64_t> &r) { result=&r; }
+
+        bool key(string_t& val) override {
+            if (val.compare("user") == 0) { user = true; }
```

Comment:
This code will do two string comparisons per key (in many instances) which is too much. Consider reducing this by doing something like...

```
  if (user) { look for id }
  else {look for user} 
```

Or something of the sort. By mindful that string comparisons are probably 'expensive'.